docs/developer.md

```diff
@@ -0,0 +1,205 @@
+# Development Workflow
+
+## Table of contents
+
+- [Prerequisites](#prerequisites)
+- [Initial Setup](#initial-setup)
+- [Development](#development)
+- [Dependencies](#dependencies)
+- [Submitting Your Changes](#submitting-your-changes)
+
+## Prerequisites
+
+* You have Go 1.10+ installed on your local host/development machine.
```

Comment:
Go version that is used in the [build file](https://github.com/openebs/dynamic-localpv-provisioner/blob/develop/buildscripts/provisioner-localpv/provisioner-localpv.Dockerfile#L15) should be used.